src/coreclr/debug/createdump/dumpwriterelf.cpp

```diff
@@ -60,26 +58,32 @@ DumpWriter::WriteDump()
     }
     else {
         ehdr.e_phnum = PH_HDR_CANARY;
+        ehdr.e_shoff = sizeof(Ehdr);
+        ehdr.e_phoff = sizeof(Ehdr) + sizeof(Shdr);
+        ehdr.e_shentsize = sizeof(Shdr);
     }
 
     if (!WriteData(&ehdr, sizeof(Ehdr))) {
         return false;
     }
 
-    size_t offset = sizeof(Ehdr) + sizeof(Shdr) + (phnum * sizeof(Phdr));
+    size_t offset = sizeof(Ehdr) + (phnum * sizeof(Phdr));
     size_t filesz = GetProcessInfoSize() + GetAuxvInfoSize() + GetThreadInfoSize() + GetNTFileInfoSize();
 
-    // Add single section containing the actual count
-    // of the program headers to be written.
-    Shdr shdr;
-    memset(&shdr, 0, sizeof(shdr));
-    shdr.sh_info = phnum;
-    // When section header offset is present but ehdr section num = 0
-    // then is is expected that the sh_size indicates the size of the
-    // section array or 1 in our case.
-    shdr.sh_size = 1;
-    if (!WriteData(&shdr, sizeof(shdr))) {
-        return false;
+    if (ehdr.e_phnum == PH_HDR_CANARY)
+    {
+        // Add single section containing the actual count of the program headers to be written.
+        Shdr shdr;
+        memset(&shdr, 0, sizeof(shdr));
+        shdr.sh_info = phnum;
+        shdr.sh_size = 1;
```

Comment:
Is there a reason we set the size to 1 here? The section address is 0. Should we mark this as `SHT_NOBITS` and set size to 0?

Reply:
It needs to be 1. 0 doesn't work.  I don't remember (and I looked again) where I found this but it needs to be 1.